From 0927c4df88664f9b2c391af9eefa303be070563f Mon Sep 17 00:00:00 2001
From: Murala S Swamy <ssmurala@habana.ai>
Date: Fri, 13 Jun 2025 18:31:10 +0300
Subject: [PATCH] [JSW-670] jgs: add arc as default arch

From JGS onwards, we dont have arm in soc.
Hence, assign arc as default type if project
is jgs.

Change-Id: I2c06e2e22bd0b05ccba8f0e027ace441cf1b092b
---

diff --git a/app/palad_bloader/Makefile b/app/palad_bloader/Makefile
index 81b76fd..88d1c16 100644
--- a/app/palad_bloader/Makefile
+++ b/app/palad_bloader/Makefile
@@ -1,7 +1,14 @@
 CC = $(CROSS_COMPILE)gcc
 OBJCOPY = $(CROSS_COMPILE)objcopy
-BASE_DEV_NAME := $(shell echo $(HL_DEV) | cut -d'_' -f1)
-HL_DEV_SUBTYPE := $(shell echo $(HL_DEV) | cut -d'_' -s -f2,3)
+
+ifeq ($(HL_DEV),jgs)
+	BASE_DEV_NAME := $(HL_DEV)
+	HL_DEV_SUBTYPE := arc
+else
+	BASE_DEV_NAME := $(shell echo $(HL_DEV) | cut -d'_' -f1)
+	HL_DEV_SUBTYPE := $(shell echo $(HL_DEV) | cut -d'_' -s -f2,3)
+endif
+
 PREBOOT_ROOT_DIR := ../..
 HL_DEV_ARCH := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/$(HL_DEV_SUBTYPE)
 HL_DEV_ARCH_COMMON := $(PREBOOT_ROOT_DIR)/arch/$(BASE_DEV_NAME)/common
